sky/backends/backend_utils.py

```diff
@@ -2390,10 +2394,15 @@ def check_cluster_available(
     bright = colorama.Style.BRIGHT
     reset = colorama.Style.RESET_ALL
     if handle is None:
+        if previous_cluster_status is None:
+            error_msg = f'Cluster {cluster_name!r} does not exist.'
+        else:
+            error_msg = (f'Cluster {cluster_name!r} was preempted or manually '
```

Comment:
Is it possible to make this more fine-grained:
- check if the cluster is a spot cluster, and if so += "was preempted" 
- check if the cluster had autodown specified, and if so += "was autodowned"
- += "or manually terminated"

Reply:
Done. Thanks!